ui/src/components/app-form/app-form.tsx

```diff
@@ -65,10 +75,173 @@ export const AppForm = ({ id }: AppFormProps): React.ReactElement => {
+  const [isUrlValid, setIsUrlValid] = useState<boolean>(false); // Track if the URL is valid
+  const [isFetching, setIsFetching] = useState<boolean>(false); // For loading state
+  const [isProcessing, setIsProcessing] = useState(false);
+  const [openModal, setOpenModal] = useState(false); // State to control modal visibility
+
+  const [repoData, setRepoData] = useState<RepoData | null>(null); // Store fetched repo data
+  const [isCondaYamlEnabled, setIsCondaYamlEnabled] = useState(false);
+  const [error, setError] = useState<string | null>(null); // Store validation errors
+  const [shouldValidate, setShouldValidate] = useState(false); // To control validation trigger
+  const repoUrlRef = useRef<HTMLInputElement>(null); // Ref for Git Repository URL field
+  const {
+    control,
+    handleSubmit,
+    setFocus,
+    setValue,
+    reset,
+    watch,
+    formState: { errors, isDirty, isValid },
+  } = useForm<AppFormInput>({
+    defaultValues: {
+      display_name: '',
+      description: '',
+      framework: '',
+      thumbnail: '',
+      filepath: initialFilepath,
+      conda_env: '',
+      custom_command: '',
+      profile: '',
+      is_public: false,
+      keep_alive: false,
+    },
+  });
+  // Use useEffect to log currentUser when the component renders or the user changes
+  useEffect(() => {
+    if (currentUser) {
+      console.log('Current User from Recoil:', currentUser); // Log the user if available
+    } else {
+      console.log('No current user available'); // Log when no user is available
+    }
+  }, [currentUser]); // Add currentUser as a dependency to trigger when it changes
+
+  useEffect(() => {
+    // Reset form fields when deployOption changes
+    reset({
+      display_name: '',
+      description: '',
+      framework: '',
+      thumbnail: '',
+      filepath: '',
+      conda_env: '',
+      custom_command: '',
+      profile: '',
+      is_public: false,
+      keep_alive: false,
+    });
+
+    // Reset additional state if necessary
+    setDescription('');
+    setRepoData(null);
+    setIsUrlValid(false);
+  }, [deployOption, reset]);
+
+  useEffect(() => {
+    const filepathFromQuery = decodeURIComponent(
+      searchParams.get('filepath') || '',
+    );
+    setValue('filepath', filepathFromQuery); // Update the form field value
+  }, [searchParams, setValue]);
+
+  const labelStyle = {
+    fontSize: '1rem',
+    transform: 'translate(14px, -6px) scale(0.75)',
+    color: 'rgba(0, 0, 0, 0.87)', // Adjust color based on your needs
+    top: '-2px',
+    left: '-5px',
+    padding: '0 4px',
+    zIndex: 1,
+    backgroundColor: '#fafbfc', // Set white background
+    paddingRight: '4px',
+    paddingLeft: '4px',
+  };
+  // Validate the URL format and fetch repo data
+  const validateGitUrl = async () => {
+    // Trigger validation after the button is clicked
+    setShouldValidate(true);
+
+    const gitHubUrlPattern =
+      /^(https?:\/\/)?(www\.)?github\.com\/[\w-]+\/[\w-]+$/;
```

Comment:
Git repo url is the clone url, this could be any valid git clone url (also not necessarily from github), for example:

- https://github.com/nebari-dev/jhub-apps-from-git-repo-example.git
- https://gitlab.com/gitlab-org/gitlab.git

Reply:
Updated the pattern to be more inclusive:
 const gitRepoUrlPattern =
      /^(https?:\/\/)?[\w.-]+\/[\w-]+\/[\w-]+(\.git)?$/i;